### PR TITLE
refactor: simplify Node.js event loop with `SpinEventLoop`

### PR DESCRIPTION
**Pull request by codebytere**
#### Description of Change

Take advantage of Node.js-provided helper function to spin uv loop found [here](https://github.com/nodejs/node/blob/eaca13ddc9092e831b9b45180c6e0ee1f3292e5e/src/api/embed_helpers.cc#L17-L60). Allows us to remove a chunk of code and maintain tighter parity with upstream.

#### Checklist

- [x] PR description included and stakeholders cc'd
- [ ] `npm test` passes
- [x] relevant documentation is changed or added
- [x] [PR release notes](https://github.com/electron/clerk/blob/master/README.md) describe the change in a way relevant to app developers, and are [capitalized, punctuated, and past tense](https://github.com/electron/clerk/blob/master/README.md#examples).

#### Release Notes

Notes: none

### CODE REVIEW

**Comment by MarshallOfSound** [comment]
@codebytere That link doesn't work, is this an exact line-for-line equivalency, or is there a chance of behavior change here?

**Comment by RaisinTen** [comment]
@MarshallOfSound I think the correct link is https://github.com/nodejs/node/blob/eaca13ddc9092e831b9b45180c6e0ee1f3292e5e/src/api/embed_helpers.cc#L17-L60.

File: shell/app/node_main.cc [hunk 1]
```diff
@@ -249,39 +249,13 @@ int NodeMain(int argc, char* argv[]) {
...
-        more = uv_loop_alive(env->event_loop());
-      } while (more && !env->is_stopping());
-      env->performance_state()->Mark(
-          node::performance::NODE_PERFORMANCE_MILESTONE_LOOP_EXIT);
-    }
-
-    env->set_trace_sync_io(false);
-
-    exit_code = node::EmitExit(env);
+    exit_code = node::SpinEventLoop(env).FromMaybe(1);
```
Comment (nornagon) [inline shell/app/node_main.cc:258]: what does a `None` result mean here? Also JS exception?

Comment (codebytere) [reply shell/app/node_main.cc:265]: added to code comments

**Review by zcbenz — APPROVED** [review]
Checked the code and they seem to be doing the exact same thing.

**Pull request merged**